lib/std/fs/test.zig

```diff
@@ -40,6 +40,39 @@ test "readAllAlloc" {
     testing.expectError(error.FileTooBig, file.readAllAlloc(testing.allocator, file_size, write_buf.len - 1));
 }
 
+test "directory operations on files" {
+    var tmp_dir = tmpDir(.{});
+    defer tmp_dir.cleanup();
+
+    const test_file_name = "test_file";
+
+    var file = try tmp_dir.dir.createFile(test_file_name, .{ .read = true });
+    file.close();
+
+    testing.expectError(error.PathAlreadyExists, tmp_dir.dir.makeDir(test_file_name));
+    testing.expectError(error.NotDir, tmp_dir.dir.openDir(test_file_name, .{}));
+    testing.expectError(error.NotDir, tmp_dir.dir.deleteDir(test_file_name));
+
+    if (builtin.os.tag != .wasi) {
+        // TODO: use Dir's realpath function once that exists
```

Comment:
Perhaps #5701 could be of use here (not necessarily saying to wait on it to land, just an FYI)